Backend/app/rider-platform/rider-app/Main/src/API/UI/Sos.hs

```diff
@@ -36,20 +38,45 @@ type API =
            :> TokenAuth
            :> ReqBody '[JSON] DSos.SosReq
            :> Post '[JSON] DSos.SosRes
+           :<|> "getDetails"
+             :> Capture "rideId" (Id DRide.Ride)
+             :> TokenAuth
+             :> Get '[JSON] DSos.SosDetailsRes
            :<|> Capture "sosId" (Id Sos.Sos)
              :> "status"
              :> TokenAuth
-             :> ReqBody '[JSON] DSos.SosFeedbackReq
+             :> ReqBody '[JSON] DSos.SosUpdateReq
+             :> Post '[JSON] APISuccess.APISuccess
+           :<|> Capture "sosId" (Id Sos.Sos)
+             :> "addVideo"
+             :> TokenAuth
+             :> MultipartForm Tmp DSos.SOSVideoUploadReq
+             :> Post '[JSON] DSos.AddSosVideoRes
+           :<|> "markRideAsSafe"
+             :> TokenAuth
+             :> Capture "sosId" (Id Sos.Sos)
              :> Post '[JSON] APISuccess.APISuccess
        )
 
 handler :: FlowServer API
 handler =
   createSosDetails
+    :<|> getSosDetails
     :<|> updateSosDetails
+    :<|> addSosVideo
+    :<|> markRideAsSafe
 
 createSosDetails :: (Id Person.Person, Id Merchant.Merchant) -> DSos.SosReq -> FlowHandler DSos.SosRes
-createSosDetails (personId, _) = withFlowHandlerAPI . withPersonIdLogTag personId . DSos.createSosDetails personId
+createSosDetails (personId, merchantId) = withFlowHandlerAPI . withPersonIdLogTag personId . DSos.createSosDetails personId merchantId
+
+getSosDetails :: Id DRide.Ride -> (Id Person.Person, Id Merchant.Merchant) -> FlowHandler DSos.SosDetailsRes
+getSosDetails rideId (personId, _) = withFlowHandlerAPI . withPersonIdLogTag personId $ DSos.getSosDetails rideId personId
+
+updateSosDetails :: Id Sos.Sos -> (Id Person.Person, Id Merchant.Merchant) -> DSos.SosUpdateReq -> FlowHandler APISuccess.APISuccess
+updateSosDetails sosId (personId, merchantId) = withFlowHandlerAPI . withPersonIdLogTag personId . DSos.updateSosDetails sosId (personId, merchantId)
+
+addSosVideo :: Id Sos.Sos -> (Id Person.Person, Id Merchant.Merchant) -> DSos.SOSVideoUploadReq -> FlowHandler DSos.AddSosVideoRes
+addSosVideo sosId (personId, _) = withFlowHandlerAPI . withPersonIdLogTag personId . DSos.addSosVideo sosId personId
 
-updateSosDetails :: Id Sos.Sos -> (Id Person.Person, Id Merchant.Merchant) -> DSos.SosFeedbackReq -> FlowHandler APISuccess.APISuccess
-updateSosDetails sosId (personId, _) = withFlowHandlerAPI . withPersonIdLogTag personId . DSos.updateSosDetails sosId personId
+markRideAsSafe :: (Id Person.Person, Id Merchant.Merchant) -> Id Sos.Sos -> FlowHandler APISuccess.APISuccess
+markRideAsSafe (personId, merchantId) sosId = withFlowHandlerAPI $ withPersonIdLogTag personId $ DSos.markRideAsSafe (personId, merchantId) sosId
```

Comment:
you can use  here function composition to avoid passing the params unnecessarily in function call.

`markRideAsSafe (personId, merchantId) = withFlowHandlerAPI . withPersonIdLogTag personId . DSos.markRideAsSafe (personId, merchantId)`